### PR TITLE
Fix an issue where flushed segments can turn into sealed

### PR DESCRIPTION
**Pull request by soothing-rain**
/kind improvement

Signed-off-by: Yuchen Gao <yuchen.gao@zilliz.com>

### CODE REVIEW

File: internal/datacoord/segment_manager.go [hunk 1]
```diff
@@ -432,7 +432,9 @@ func (s *SegmentManager) SealAllSegments(ctx context.Context, collectionID Uniqu
 		if info.CollectionID != collectionID {
 			continue
 		}
-		if info.State == commonpb.SegmentState_Sealed {
+		if info.State == commonpb.SegmentState_Sealed ||
+			info.State == commonpb.SegmentState_Flushing ||
+			info.State == commonpb.SegmentState_Flushed {
```
[the remark this answers is not in the document]
Comment (soothing-rain) [reply internal/datacoord/segment_manager.go]: Done.

**Review by congqixia — APPROVED** [review]
/lgtm

**Comment by soothing-rain** [comment]
rerun ut

**Pull request closed without merging**